llvm/cmake/modules/TableGen.cmake

```diff
@@ -92,8 +92,23 @@ function(tablegen project ofn)
   get_directory_property(tblgen_includes INCLUDE_DIRECTORIES)
   list(TRANSFORM tblgen_includes PREPEND -I)
 
+  set(tablegen_exe ${${project}_TABLEGEN_EXE})
+  set(tablegen_depends ${${project}_TABLEGEN_TARGET} ${tablegen_exe})
+  if (LLVM_TABLEGEN_USE_RELEASE_BUILD)
```

Comment:
This is something that already is supported without the multi-generator support `LLVM_OPTIMIZED_TABLEGEN`.  I'd rather key off the same option and change the implementation if we are targeting a multi-generator build.

Reply:
Good idea!